CHANGELOG.md

```diff
@@ -3,7 +3,163 @@
+- ast/compile: Sort modules by file name in NewModuleTree (#5057) ([#5056](https://github.com/open-policy-agent/opa/issues/5056))
+- ast/compile: Respect unsafeBuiltinMap for 'with' replacements (#5107) (authored by @srenatus)
+- ast/compile: Return immediately if an "after" stage failed (#5063) (authored by @srenatus)
+- ast/term.go: copy() nitpicks (#5058) (authored by @srenatus)
+- ast: Add cap to array initialization when known (#5037) (authored by @mstrYoda)
+- topdown/object: Fix unchecked err in object.union_n (#5074) ([#5073](https://github.com/open-policy-agent/opa/issues/5073)) authored by @charlesdaniels reported by @charlesdaniels
+- topdown/reachable: Fix missing operand type checks. (#4956) ([#4951](https://github.com/open-policy-agent/opa/issues/4951)) authored by @philipaconrad
+- topdown/type+wasm: Fix inconsistent is_<type> return values. (#5064) ([#4943](https://github.com/open-policy-agent/opa/issues/4943)) authored by @philipaconrad reported by @ericjkao
+- Add query param in canonical request of aws Sigv4 signature to avoid 403 from aws (#4977) (authored by @sinhaaks)
+
+#### Test Suite
+
+- Add a few missing tests (#4957)
+- topdown/units_parse: Avoid extra decimal places for integers. (#5011)
+- Add error type to units.* builtin test assertions (#4966)
+- test/e2e/certrefresh: Add file.Sync() to eliminate test failures due to slow disk writes. (#5033)
+- testcases: Fix duplicate yaml test keys (#4970)
+- topdown/exported_tests: Remove golang 1.16 x509 exception (#5032)
+- wasm/tests/test: Fix broken WASM C tests for is_type builtins. (#5075)
+- cmd/bench: Fix port collision in utility function used for E2E testing.
+
+### Documentation
+
+- SECURITY: Migrate policy to web site, update content (#5094) ([#4272](https://github.com/open-policy-agent/opa/issues/4272)) reported by @adoliver
+- Add deprecated flag to all deprecated builtins (#5078) ([#5072](https://github.com/open-policy-agent/opa/issues/5072))
+- builtins: Update description of format_int to say it rounds down
+- docs/policy-reference: Update Rego EBNF grammar (#4955) (authored by @shaded-enmity)
+- docs/builtins: Fix typo in semver.compare (#5059) ([#5012](https://github.com/open-policy-agent/opa/issues/5012)) reported by @tetsuya28
+- docs: Fix AWS Signature section in Configuration (authored by @pauly4it)
+- docs: Update port and bundle folder for GraphQL tutorial
+- docs: Document that function overloading is unsupported
+- docs: Fixing related_resources annotations example ([#4982](https://github.com/open-policy-agent/opa/issues/4982)) reported by @humbertoc-silva
+- docs: Fixing typo in metadata, issue #5018 (#5019) ([#5018](https://github.com/open-policy-agent/opa/issues/5018)) authored by @cimin0 reported by @cimin0
+
+### Website + Ecosystem
+
+- Update links to opa-kafka-plugin (#5021)
+- Add OCI documentation (#4781) (authored by @carabasdaniel)
+- Fix broken link (#5071) (authored by @unautre)
```

Comment:
Can you add more context here.

Reply:
That's a particularly egregious commit message; I'll flesh it out shortly.

Reply:
It is done. :+1: